source/lambda/documentbulkprocessor/lambda_function.py

```diff
@@ -12,16 +12,177 @@
+    # try to fetch amd load the policy
+    try:
+        
+        policyData = s3helper.readFromS3(ingestionBucketName,
+                                       policyKey,
+                                       os.environ['AWS_REGION'])
+
+
+    # the normal case of a file not provided is handled.  If any other error
+    # occur the indexing will proceed without the membership tags in the policy file
+    except ClientError as e:
+        policyData = None
+        # NoSuchKey is the expected exception when no policy provided
+        if e.response['Error']['Code'] == 'NoSuchKey':
+            print("no kendra policy file found, skipping")
+        else:
+            print("ClientError exception from s3helper.readFromS3() for policy file: " + str(e))
 
-        print("event: {}".format(event))
+    # an error that should be investigated
     except Exception as e:
-        print("Failed to process records. Exception: {}".format(e))
+        policyData = None
+        print("unspecified exception from s3helper.readFromS3() for policy file: " + str(e))
+                                    
+    # generate UUID for document
+    documentId = generateDocumentID(os.environ['OUTPUT_BUCKET'], s3Client)
+
+    # upload document in document bucket
+    destinationDocumentKey = "public/" + documentId + "/" + ingestionDocumentFilename
+
+    ret = uploadDocument(os.environ['OUTPUT_BUCKET'],
+                         destinationDocumentKey,
+                         ingestionDocumentFilename,
+                         s3Client)
+
+    # in case of error uploading document, we exit
+    if ret == False:
+        print("failed to upload document into output bucket: " + destinationDocumentKey)
+        return
+
+    # if optional Kendra policy was present, upload it to document folder
+    # alongside document
+    if policyData != None:
+        print("copying over to document folder the kendra policy file")
+        s3helper.writeToS3(policyData,
+                           os.environ['OUTPUT_BUCKET'],
+                           "public/" + documentId + "/" + policyFilename,
+                           awsRegion=os.environ['AWS_REGION'])
+
+    # start normal document processing by creating a DynamoDB record,
+    # another lambda function will pick it up from a DynamoDB stream
+    # event
+    docStore = DocumentStore(os.environ['DOCUMENTS_TABLE'],
+                             os.environ['OUTPUT_TABLE'])
+
+    docStore.createDocument(documentId,
+                            os.environ['OUTPUT_BUCKET'],
+                            destinationDocumentKey)
+
+    return
+
+
+def processQueueRecord(queueRecord):
+
+    records = json.loads(queueRecord['body'])
+
+    # one of more s3 event records
+    if 'Records' in records:
+        for record in records['Records']:
+            try:
+                if record['eventSource'] == 'aws:s3':
+                    if record['eventName'] == 'ObjectCreated:Put' or record['eventName'] == 'ObjectCreated:Copy':
```

Comment:
The infrastructure stack has been defined only to trigger the lambda on PUT events : https://github.com/awslabs/document-understanding-solution/blob/kendraMaster/source/lib/cdk-textract-stack.ts#L771
Do we need still need to check the event type?

Reply:
i saw  "Test" events go by in the early moments of deploy,  it's just safer to check and log, in case someone else changes this without looking at this code here